### ISSUE TITLE
Kubernetes namespace is undeletable after 3rd party resources are created

### ISSUE THREAD

**Issue by kyessenov** [istio/manager#15]
Due to upstream bug https://github.com/kubernetes/kubernetes/issues/37278.
Logs:
E1217 00:05:43.750418    2919 errors.go:63] apiserver received an error that is not an unversioned.Status: no kind "DeleteOptions" is registered for version "istio.io/v1"
E1217 00:05:43.750723    2919 namespace_controller.go:176] no kind "DeleteOptions" is registered for version "istio.io/v1"

**Comment by GregHanson** [comment]
@kyessenov I see they have closed that issue and if I read it right, the suggested workaround is to delete all the resources before deleting the namespace.  Though apparently this solution is "not a silver bullet . . . use at your own discretion"

**Comment by kyessenov** [comment]
My understanding is that TPRs are shared across namespaces. So we cannot delete TPRs per namespace. The only workaround is to go under kube API and manipulate them directly in etcd.

**Comment by kyessenov** [comment]
Quick clarification - I mean TPR declarations, not TPRs themselves.

**Comment by GregHanson** [comment]
@zcahana @elevran: I think one or both of you have some experience working with third party resources in Kubernetes.  Any thoughts on this issue?

**Comment by ayj** [comment]
This seems to be fixed in 1.5.2.

```
$ kubectl create namespace mynamespace
namespace "mynamespace" created

$ kubectl create -f resource.yaml
thirdpartyresource "cron-tab.stable.example.com" created

$ kubectl delete namespace mynamespace
namespace "mynamespace" deleted

$ kubectl get namespace
NAME          STATUS        AGE
default       Active        3m
kube-system   Active        3m
mynamespace   Terminating   31s           # <=== 1.5.1 gets stuck here

[15:26:00] jasonyoung@ayj0:~
$ kubectl get namespace
NAME          STATUS    AGE
default       Active    3m
kube-system   Active    3m

**Comment by kyessenov** [comment]
Thanks for the heads up!
